app/api/routers/breeze_buddy/knowledge_base/handlers.py

```
@@ -0,0 +1,373 @@
+    kb = await _get_kb_or_404(kb_id)
+    validate_kb_access(current_user, kb.reseller_id, kb.merchant_id, "query")
+
+    started = time.perf_counter()
+    try:
+        chunks = await retrieve(
+            kb_ids=[kb_id],
+            query=request.query,
+            top_k=request.top_k,
+            score_threshold=request.score_threshold,
+        )
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)